workflows/cloudify_system_workflows/snapshots/snapshot_restore.py

```diff
@@ -117,14 +110,55 @@ def restore(self):
                 self._restore_influxdb()
                 self._restore_credentials(postgres)
                 self._restore_agents()
-                self._restore_deployment_envs(existing_dep_envs)
                 self._restore_amqp_vhosts_and_users()
+                self._restore_deployment_envs()
             if self._restore_certificates:
                 self._restore_certificate()
         finally:
             ctx.logger.debug('Removing temp dir: {0}'.format(self._tempdir))
             shutil.rmtree(self._tempdir)
 
+    def _restore_deployment_envs(self):
+        deps = {}
+        for tenant in self._client.tenants.list():
+            ctx.logger.info(tenant)
+            tenant_name = tenant['name']
+            ctx.logger.info(tenant_name)
+            ctx._context['tenant']['original_name'] = ctx.tenant_name
+            ctx._context['tenant']['name'] = tenant_name
+            ctx._dep_contexts = None
+            deps[tenant_name] = ctx.deployments_contexts
+            ctx._context['tenant']['name'] = (
+                ctx._context['tenant']['original_name']
+            )
+            ctx._context['tenant'].pop('original_name')
+
+        for tenant, deployments in deps.iteritems():
+            ctx.logger.debug(
```

Comment:
Info makes sense for this, I think, yes. Especially as it only appears in /var/log/mgmtworker/cloudify/logs/__system__.log

(which is something I'd like to fix but I don't think there's time for this release)